src/main/g8/build.sbt

```diff
@@ -14,10 +14,33 @@ lazy val root = (project in file("."))
     // Options used by `native-image` when building native image.
     // https://www.graalvm.org/docs/reference-manual/native-image/
     graalVMNativeImageOptions ++= Seq(
-        "--initialize-at-build-time", // Auto-packs dependent libs at build-time
-        "--no-fallback", // Bakes-in run-time reflection (alternately: --auto-fallback, --force-fallback)
-        "--no-server", // Won't be running `graalvm-native-image:packageBin` often, so one less thing to break
-        "--static" // Disable for OSX (non-docker) builds - Forces statically-linked binary, compatible with libc (linux)
+      "--initialize-at-build-time", // Auto-packs dependent libs at build-time
+      "--no-fallback", // Bakes-in run-time reflection (alternately: --auto-fallback, --force-fallback)
+      "--no-server" // Won't be running `graalvm-native-image:packageBin` often, so one less thing to break
+    ),
+    // Additional options used by `native-image` when building native image that vary based on the
+    // operating system of the machine that is building the image
+    graalVMNativeImageOptions ++= {
+      // Note: Ideally, we should be checking if the current system has a working libc implementation here,
+      // but os.name works surprisingly well in most cases.
+      System.getProperty("os.name").toLowerCase match {
+        case mac if mac.contains("mac") =>
+          Nil
+        case win if win.contains("win") =>
+          Nil
+        case _ =>
+          // Forces statically-linked binary, compatible with libc (linux)
+          Seq("--static")
+      }
+    },
+
+    // Contains substitutions that are necessary for native image creation. These substitutions are required since monix
+    // depends on JCTools which contains a lot of code that can't be compiled by graal native:
+    libraryDependencies ++= Seq(
+      "science.doing" % "jctools-graal-native_20_2_0" % "1.0.0",
+      // Important: Keep the version of the "org.graalvm.nativeimage:svm" dependency synced with whatever version of
+      // graal we are using!
+      "org.graalvm.nativeimage" % "svm" % "20.2.0" % Provided
```

Comment:
Shameless integration of my own lib into this PR :rofl:

Reply:
Don't even START to apologize on this one. It was shameless. As for the "perhaps GraalVM and Monix" already have a patch for this, I can tell you with almost 99.99% confidence they don't yet.

Monix is still unfortunately still using `sun.misc.Unsafe` via their `JCTools` underbelly. I've been watching the GraalVM releases like a hawk waiting for an update to be able to auto-substitute it and have yet to see anything. That's not to say they haven't patched a HUGE number of issues of course, but yeah, no dice.

**HOWEVER**, this matters not! If there is one thing I've learned about Graal Native, it's that sharing substitutions via libraries is a heck of a lot better of an idea on paper than it is in practice for two reasons:
1. By design, it obfuscates what pieces of the code are being substituted, and as such, which makes it really hard to tell if/when a substitution is no longer needed. It also contributes even more to the "voodoo magic-ness" of a project.
2. Versioning substitutions is just a pain in the butt in general. It's FAR easier to just copy the damn files around. As crazy as that may sound.

That brings up an interesting question: **Do you wanna include substitutions in this project?** On one hand, it makes your project a little bit fugly. Naturally, folks are gonna see your substitution files and either go "WTF" or at the very least "I don't think I'm in Kansas anymore". On the other, it shows folks that substitutions are INDEED an actual thing.

**So moving forward, I got three ideas, all with their own pros/cons:**
A) Kill the dependency on my crappy lib, move the substitutions into this project, and call it a day.
B) Kill the dependency on my crappy lib, remove the substitutions, remove Observable (which is what actually uses JCTools in the first place).
C) Kill the dependency on my crappy lib, remove the substitutions, replace `monix` with `cats-effect` (here is where you say: "say whaaaaaat?") or even `ZIO` (for the record though: I've yet to use ZIO personally).

Alright, alright, lemme justify option C a bit more, cause it's a :fire: take:
1. `cats-effect 3.x.x` is super easy to use (unfortunately, can't say the same for 2.x), and has YET to let me down when it comes to packing it into a native image. 
2. It means that when folks EXPAND on this code and start using new features, there is a significantly reduced chance of them running into a problem that requires substitutions to solve. If you keep `monix` and/or the substitutions, there is still a very high chance of someone calling a different part of monix's library, and consequently allowing the image process to discover a whole new area of code to run into problems with.

**TL;DR**: Rock, meet hard place. I don't like ANY of the options I presented, maybe you can find a fourth?

Reply:
> I ain't. I'm apologizing for how damn long it has taken me to get to reviewing this dope PR.

:joy: :joy: :joy: All gucci, my dude! Life happens.

Reply:
> C compelling. 

Here it is if you wanna see what it could look like: https://github.com/sbchapin/quickstart-scala-sbt-native-image.g8/pull/5